modules/optimization/src/functions/VectorNearestPointFunction.C

```diff
@@ -0,0 +1,197 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "VectorNearestPointFunction.h"
+
+registerMooseObject("OptimizationApp", VectorNearestPointFunction);
+
+InputParameters
+VectorNearestPointFunction::validParams()
+{
+  InputParameters params = OptimizationFunction::validParams();
+  params.addClassDescription(
+      "Function based on the nearest point to coordinates and values defined by a vector "
+      "values, interpolates linearly in time with transient data.");
+  params.addParam<ReporterName>(
+      "coord_x",
+      "Vector value containing x-coordinate of points, default is assumed to be all 0s.");
```

Comment:
these arent vectors in the usual meaning, they are vectors from VPPs

Reply:
description should say "Reporter vector" maybe

not sure what is clear enough

Reply:
done